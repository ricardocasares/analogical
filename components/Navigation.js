import Link from "./Link";

export default () => (
  <nav>
    <ul>
      <li>
        <Link prefetch route="index">
          Home
        </Link>
      </li>
      <li>
        <Link prefetch route="work">
          Work
        </Link>
      </li>
      <li>
        <Link prefetch route="/changelog">
          Changelog
        </Link>
      </li>
      <li>
        <Link prefetch route="/photos">
          Photos
        </Link>
      </li>
    </ul>
    <style jsx>{`
      nav {
        margin-top: 20px;
      }

      ul {
        list-style: none;
        margin: 0;
        padding: 0;
      }

      li {
        display: inline;
        margin-right: 20px;
      }

      li:last-of-type {
        margin-right: 0;
      }

      @media (min-width: 425px) {
        nav {
          margin-top: 0;
          text-align: right;
        }
      }
    `}</style>
  </nav>
);
