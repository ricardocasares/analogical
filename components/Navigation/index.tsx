import Link from "@app/components/Link";
import HorizontalScroll from "@app/components/Scroller";
import { NavList } from "@app/components/Typography/List";
import Logo from "@app/components/Logo";

export const Navigation = () => (
  <HorizontalScroll>
    <nav>
      <NavList>
        <li>
          <Link href="/">
            <a>
              <Logo />
            </a>
          </Link>
        </li>
        <li>
          <Link href="/changelog" partial>
            <a>Changelog</a>
          </Link>
        </li>
      </NavList>
    </nav>
  </HorizontalScroll>
);

export default Navigation;
