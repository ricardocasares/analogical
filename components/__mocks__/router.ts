import Router from "next/router";

// @ts-ignore
Router.router = {
  push: () => {},
  prefetch: () => {}
};
