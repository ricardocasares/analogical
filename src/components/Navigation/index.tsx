import Link from "next/link";
import { Logo } from "@/components/Logo";
import { Toggle } from "@/components/Toggle";
import { A, NavList } from "@/components/Typography";
import { HorizontalScroll } from "@/components/HorizontalScroll";

export const Navigation = () => {

  return (
    <HorizontalScroll>
      <nav>
        <NavList css={{ display: 'flex' }}>
          <li>
            <Link href="/" passHref>
              <A><Logo /></A>
            </Link>
          </li>
          <li>
            <Link href="/changelog" passHref>Changelog</Link>
          </li>
          <li>
            <Link href="/code" passHref>Code</Link>
          </li>
          <li><Toggle /></li>
        </NavList>
      </nav>
    </HorizontalScroll>
  );
};
