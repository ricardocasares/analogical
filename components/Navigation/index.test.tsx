import React from "react";
import renderer from "react-test-renderer";
import serializer from "jest-emotion";
import Navigation from "@app/components/Navigation";
import "../__mocks__/router";

expect.addSnapshotSerializer(serializer);

test("renders with correct styles", () => {
  expect(renderer.create(<Navigation />).toJSON()).toMatchSnapshot();
});
