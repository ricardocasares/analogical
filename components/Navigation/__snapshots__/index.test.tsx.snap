// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with correct styles 1`] = `
.emotion-4 {
  overflow-x: scroll;
  overflow-y: hidden;
  white-space: nowrap;
  -webkit-overflow-scrolling: touch;
}

.emotion-4::-webkit-scrollbar {
  display: none;
}

.emotion-2 {
  margin: 0;
  padding: 0;
  list-style: none;
}

.emotion-2 li {
  display: inline-block;
  margin-right: 10px;
}

.emotion-2 li:last-child {
  margin-right: 0;
}

.emotion-2 a {
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: lowercase;
  -webkit-transition: color 1s;
  transition: color 1s;
  font-size: 16px;
}

.emotion-1 {
  font-weight: 500;
  text-transform: lowercase;
  color: currentcolor;
}

.emotion-0 {
  display: none;
}

@media screen and (min-width:40em) {
  .emotion-0 {
    display: inline;
  }
}

<div
  className="emotion-4"
>
  <nav>
    <ul
      className="emotion-2 emotion-3"
    >
      <li>
        <a
          className={null}
          href="/"
          onClick={[Function]}
        >
          <span
            className="emotion-1"
          >
            ùå∫ 
            <div
              className="emotion-0"
              display={
                Array [
                  "none",
                  "inline",
                ]
              }
            >
              analogic.al
            </div>
          </span>
        </a>
      </li>
      <li>
        <a
          className={null}
          href="/changelog"
          onClick={[Function]}
        >
          Changelog
        </a>
      </li>
      <li>
        <a
          className={null}
          href="/projects"
          onClick={[Function]}
        >
          Projects
        </a>
      </li>
    </ul>
  </nav>
</div>
`;
